core/handler/convert_fields.go

```diff
@@ -240,6 +240,11 @@ func (h *handler) ConvertFieldsDown(ctx log.Interface, appDown *mqtt.DownlinkMes
 		return nil
 	}
 
+	// Impossible to have fields and payload at the same time
+	if appDown.Payload != nil {
+		return errors.New("Error processing downlink: both Fields and Payload provided")
```

Comment:
We're not "processing downlink" here, it's converting fields down. The caller knows the method it's calling, so just say that both fields and payload are set, and that that's not possible.